extensions/amp-lightbox-viewer/0.1/service/lightbox-manager-impl.js

```diff
@@ -83,23 +102,39 @@ export class LightboxManager {
    */
   scanLightboxables_() {
     return this.ampdoc_.whenReady().then(() => {
+      const viewer = elementByTag(this.ampdoc_.getRootNode(), VIEWER_TAG).id;
       const matches = this.ampdoc_.getRootNode().querySelectorAll('[lightbox]');
-      this.elements_ = [];
       for (let i = 0; i < matches.length; i++) {
```

Comment:
For iterating over results from `querySelectorAll`, use [`src/dom.js#iterateCursor`](https://github.com/ampproject/amphtml/blob/master/src/dom.js#L621) for performance

Reply:
[40% difference?! Seriously....](https://jsperf.com/iterating-over-collections-of-elements). WOW.

Reply:
Right?? I *think* it's because an iterator is used under the hood, so evaluating `elements[i]` and `elements.length` is more expensive than you might expect.